hare/algorithm.go

```diff
@@ -52,12 +52,12 @@ const (
 	notCompleted = false
 )
 
-// procReport is the termination report of the CP.
-// It consists of the layer id, the set we agreed on (if available) and a flag to indicate if the CP completed.
+// procReport is the termination report of the CP
 type procReport struct {
-	id        instanceID
-	set       *Set
-	completed bool
+	id        instanceID // layer id
+	set       *Set       // agreed-upon set
+	coinflip  bool       // weak coin value
+	completed bool       // whether the CP completed
```

Comment:
renaming the fields would be nicer 😢

Reply:
You mean rather than adding the explanatory comments?